src/command_modules/azure-cli-role/azure/cli/command_modules/role/_help.py

```diff
@@ -9,6 +9,7 @@
 helps['ad sp create-for-rbac'] = """
     type: command
     short-summary: Create a service principal and configure its access to Azure resources.
+    Note: if cli login is using service principal, the service principal should have read and write access to Microsoft Graph appliation in the AAD tenant.
```

Comment:
Could you cross check whether this is all needed? Based on my experience, 2 things need to happen
1.  The permission should be granted to `Windows Azure Active Directory`, rather `Microsoft Graph`. These are 2 different resources
2. Consent the permission required by the service principal. If you are in portal, you can just click the "grant permission" button

If this is what you went through to get things work as well, could you update the text here?